File: DataStructures/implementations/src/main/java/datastructures/Queue.java
package datastructures;

public class Queue<T> {
  private DoubleLinkedList<T> linkedList = new DoubleLinkedList<>();
  public void insert(T data) {
    linkedList.append(data);
  }

  public T first() {
    return linkedList.getFirst().data;
  }

  public T removeFirst() {
    return linkedList.deleteFirst().data;
  }

  public boolean isEmpty() {
    return linkedList.isEmpty();
  }
}

File: go/data_structures/linked_list/queue/queue.go
package queue

import . "github.com/alexsotocx/algorithms/go/data_structures/linked_list"

type Queue interface {
	PushBack(data interface{}) (bool, error)
	DeleteHead() (interface{}, error)
	First() interface{}
	Iterator() IteratorInterface
	IsEmpty() bool
}

func New() Queue {
	return &DoubleLinkedList{}
}
